railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -13,7 +13,7 @@ def initialize(app, config, paths)
 
       def build_stack
         ActionDispatch::MiddlewareStack.new do |middleware|
-          middleware.use ::ActionDispatch::HostAuthorization, config.hosts, config.action_dispatch.hosts_response_app
+          middleware.use ::ActionDispatch::HostAuthorization, hosts: config.hosts, **config.host_authorization
```

Comment:
This will allow `config.host_authorization[:hosts]` to override `config.hosts`, which seems undesirable. I think we should leave `hosts` as a positional argument.

Reply:
Hosts as a position arguments seems better to me as well, it separates the required input (the hosts) from the settings of the middleware.

Reply:
:+1: